crates/moon/tests/test_cases/mod.rs

```diff
@@ -5972,3 +5972,45 @@ fn test_js() {
         "#]],
     );
 }
+
+#[test]
+fn test_generate_test_driver_incremental() {
+    let dir = TestDir::new("moon_test_hello_lib.in");
+
+    get_stdout_with_args_and_replace_dir(&dir, ["test", "--package", "moonbitlang/hello/lib"]);
+    let driver_file =
+        dir.join("target/wasm-gc/debug/test/lib/__generated_driver_for_internal_test.mbt");
+    assert!(driver_file.exists());
+
+    let time_1 = driver_file.metadata().unwrap().modified().unwrap();
+
+    get_stdout_with_args_and_replace_dir(
+        &dir,
+        [
+            "test",
+            "--package",
+            "moonbitlang/hello/lib",
+            "--file",
+            "hello_wbtest.mbt",
+        ],
+    );
+    let time_2 = driver_file.metadata().unwrap().modified().unwrap();
+
+    assert!(time_1 == time_2);
+
+    get_stdout_with_args_and_replace_dir(
+        &dir,
+        [
+            "test",
+            "--package",
+            "moonbitlang/hello/lib",
+            "--file",
+            "hello_wbtest.mbt",
+            "--index",
+            "0",
+        ],
+    );
+    let time_3 = driver_file.metadata().unwrap().modified().unwrap();
+
+    assert!(time_2 == time_3);
```

Comment:
add a counterexample?
for example, add a new mbt file to force rebuilding, then we should have something like `time_4 != time_3`